File: test-suite/src/test/java/io/micronaut/docs/client/MyService.java
package io.micronaut.docs.client;

//tag::class[]
import io.kubernetes.client.openapi.ApiException;
import io.kubernetes.client.openapi.apis.CoreV1Api;
import io.kubernetes.client.openapi.models.V1PodList;
import jakarta.inject.Singleton;

@Singleton
public class MyService {

    private final CoreV1Api coreV1Api;

    public MyService(CoreV1Api coreV1Api) {
        this.coreV1Api = coreV1Api;
    }

    public void myMethod(String namespace) throws ApiException {
        V1PodList v1PodList = coreV1Api.listNamespacedPod(namespace).execute();
    }
}
//end::class[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/client/MyService.kt
package io.micronaut.docs.client

//tag::class[]
import io.kubernetes.client.openapi.ApiException
import io.kubernetes.client.openapi.apis.CoreV1Api
import jakarta.inject.Singleton

@Singleton
class MyService(private val coreV1Api: CoreV1Api) {

    @Throws(ApiException::class)
    fun myMethod(namespace: String) {
        val v1PodList = coreV1Api.listNamespacedPod(namespace).execute()
    }
}
//end::class[]
